components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIClientGenerationManager.java

```diff
@@ -177,16 +196,26 @@ public String sdkGeneration(String appName, String sdkLanguage, String userName,
             spec = new File(specLocation);
 
             FileWriter fileWriter = null;
+            BufferedWriter bufferedWriter = null;
             try {
                 if (!spec.exists()) {
                     spec.createNewFile();
                 }
                 fileWriter = new FileWriter(spec.getAbsoluteFile());
-                BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+                bufferedWriter = new BufferedWriter(fileWriter);
                 bufferedWriter.write(swagger);
-                bufferedWriter.close();
             } catch (IOException e) {
                 log.error("problem when storing the temporary swagger file", e);
+                throw new APIClientGenerationException("problem when storing the temporary swagger file", e);
+            } finally {
+                if (bufferedWriter != null) {
+                    try {
+                        bufferedWriter.close();
+                    } catch (IOException e) {
+                        log.error("problem when closing the connection which is opened to store the swagger file", e);
+                        throw new APIClientGenerationException("problem when closing the connection which is opened to store the swagger file", e);
```

Comment:
We don't have to throw an exception inside the finally block here. Please remove it.